eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/EidasSamlApplication.java

```diff
@@ -37,6 +53,33 @@ public void initialize(final Bootstrap<EidasSamlConfiguration> bootstrap) {
 
         VerifySamlInitializer.init();
 
+
+        connectorMetadataResolverBundle = new MetadataResolverBundle<>(EidasSamlConfiguration::getConnectorMetadataConfiguration);
+        
+        bootstrap.addBundle(connectorMetadataResolverBundle);
         bootstrap.addBundle(new LogstashBundle());
     }
+
+    @Override
+    public void run(EidasSamlConfiguration configuration, Environment environment) throws Exception {
+        EidasAuthnRequestValidator eidasAuthnRequestValidator = createEidasAuthnRequestValidator(configuration, connectorMetadataResolverBundle);
+        environment.jersey().register(new EidasSamlResource());
+    }
+
+    private EidasAuthnRequestValidator createEidasAuthnRequestValidator(EidasSamlConfiguration configuration, MetadataResolverBundle hubMetadataResolverBundle) throws Exception {
+        MessageReplayChecker replayChecker = configuration.getReplayChecker().createMessageReplayChecker("eidas-saml-parser");
+        DestinationValidator destinationValidator = new DestinationValidator(
+                configuration.getProxyNodeAuthnRequestUrl(), configuration.getProxyNodeAuthnRequestUrl().getPath());
+
+        return new EidasAuthnRequestValidator(
```

Comment:
We could consider making all these Validators implement a new EidasSAMLValidator interface with a validate(T) method, and have the resource iterate over a supplied list of `EidasSAMLValidator`s.

Reply:
We're basically reimplementing OpenSAML with our validators. What you've mentioned somewhat exists in OpenSAML as `MessageHandler` and `MessageHandlerChain`.

Reply:
Yep, we can change this later on.